### PR TITLE
Fortran parser improvements from Geany

### PR DESCRIPTION
**Pull request by techee**
These are various improvements we had in Geany for the Fortran parser:

- reporting of anonymous tags
- reporting scopes for interfaces and enums
- allowing keyword names to be used as names of various Fortran types

Note: I don't know what I'm doing much, I don't know Fortran and it's possible I'm doing something wrong - the patches are just based on what is present in Geany to allow us to use the uctags Fortran parser with correct scope information.

### CODE REVIEW

File: parsers/fortran.c [hunk 1]
```diff
@@ -479,6 +481,15 @@ static tokenInfo *newTokenFrom (tokenInfo *const token)
 	return result;
 }
 
+static tokenInfo *newAnonTokenFrom (tokenInfo *const token, unsigned int uTagKind)
+{
+	tokenInfo *result = newTokenFrom (token);
```
Comment (masatake) [inline parsers/fortran.c:485]: Could you add the following change:
```
-static tokenInfo *newTokenFrom (tokenInfo *const token)
+static tokenInfo *newTokenFromFull (tokenInfo *const token, bool noCopyStr)
 {
 	tokenInfo *result = xMalloc (1, tokenInfo);
 	*result = *token;
-	result->string = vStringNewCopy (token->string);
+	result->string = noCopyStr? vStringNewCopy (token->string): vStringNew();
 	token->secondary = NULL;
 	token->parentType = NULL;
 	token->signature = NULL;
 	return result;
 }
 
+static tokenInfo *newTokenFrom (tokenInfo *const token)
+{
+	return newTokenFromFull (token, true);
+}
+
```

So you can use `newTokenFromFull (token, false` here.
As the result you don't have to call `vStringClear (result->string);`.

Comment (masatake) [reply parsers/fortran.c:485]: Other than this minor point, all changes look fine.

Comment (techee) [reply parsers/fortran.c]: Done.

**Comment by masatake** [comment]
I will reorder the commits and merge them.

**Comment by masatake** [comment]
I wonder my pull requests are failed always on some CI platforms.
On the other hand, your pull requests are successful always on the same CI platforms.

**Comment by masatake** [comment]
Merged via #3240. Thank you.

**Comment by techee** [comment]
> I wonder my pull requests are failed always on some CI platforms.
> On the other hand, your pull requests are successful always on the same CI platforms.

I had to re-run them to make them succeed - it seems github had some problems yesterday.

**Pull request closed without merging**